share/tools/web_config/sample_prompts/sorin.fish

```diff
@@ -1,61 +1,219 @@
 # name: Sorin
 # author: Ivan Tham <ivanthamjunhoe@gmail.com>
 
+# Sources:
+# - General theme setup: https://github.com/sorin-ionescu/prezto/blob/d275f316ffdd0bbd075afbff677c3e00791fba16/modules/prompt/functions/prompt_sorin_setup
+# - Extraction of git info: https://github.com/sorin-ionescu/prezto/blob/d275f316ffdd0bbd075afbff677c3e00791fba16/modules/git/functions/git-info#L180-L441
+
 function fish_prompt
-    test $SSH_TTY
-    and printf (set_color red)$USER(set_color brwhite)'@'(set_color yellow)(prompt_hostname)' '
-    test "$USER" = 'root'
-    and echo (set_color red)"#"
+    if test -n "$SSH_TTY"
+        echo -n (set_color brred)"$USER"(set_color white)'@'(set_color yellow)(prompt_hostname)' '
+    end
 
-    # Main
-    echo -n (set_color cyan)(prompt_pwd) (set_color red)'❯'(set_color yellow)'❯'(set_color green)'❯ '
+    echo -n (set_color blue)(prompt_pwd)' '
+
+    set_color -o
+    if test "$USER" = 'root'
+        echo -n (set_color red)'# '
+    end
+    echo -n (set_color red)'❯'(set_color yellow)'❯'(set_color green)'❯ '
+    set_color normal
 end
 
 function fish_right_prompt
-    # last status
-    test $status != 0
-    and printf (set_color red)"⏎ "
-
-    if set -l git_dir (git rev-parse --git-dir 2>/dev/null)
-        # Magenta if branch detached else green
-        set -l branch (command git branch -qv | string match "\**")
-        string match -rq detached -- $branch
-        and set_color brmagenta
-        or set_color brgreen
-
-        git name-rev --name-only HEAD
-
-        # Merging state
-        test -f "$git_dir/MERGE_HEAD"
-        and printf ':'(set_color red)'merge'
-        printf ' '
-
-        # Symbols
-        if set -l count (command git rev-list --count --left-right $upstream...HEAD 2>/dev/null)
-            echo $count | read -l ahead behind
-            if test "$ahead" -gt 0
-                printf (set_color magenta)⬆' '
-            end
-            if test "$behind" -gt 0
-                printf (set_color magenta)⬇' '
-            end
+    set -l cmd_status $status
+    if test $cmd_status -ne 0
+        echo -n (set_color red)"✘ $cmd_status "
+    end
+
+    # Get the git directory for later use.
+    # Return if not inside a Git repository work tree.
+    set -l git_dir (command git rev-parse --git-dir ^&-)
+    if test $status -ne 0
+        set_color normal
+        return
+    end
+
+    # Return if zprezto's git-info is disabled.
+    set -l git_info_enabled (command git config --bool prompt.showinfo)
+    if test "$git_info_enabled" = 'false'
+        set_color normal
+        return
+    end
+
+    set -l commit ''
+    if set -l action (_git_action "$git_dir")
+        set commit (command git rev-parse --short=7 HEAD 2> /dev/null)
```

Comment:
See 8e88b29eeb9aa7c42a00a4188c6c26a17d8ad9f5. Git's shortening can be quite slow, because it tries to be unambiguous. For a prompt, that's overkill.